### PR TITLE
[Gen 2][Mobile] Update social sign-in to support iOS, Android, Flutter and React Native.

### PR DESCRIPTION
**Pull request by salihgueler**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [x] Swift
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 1]
```diff
@@ -303,6 +323,49 @@ export const auth = defineAuth({
...
+
+To enable this capability, an additional dependency must be installed.
+
+```sh
+npm install @aws-amplify/rtn-web-browser
+```
+
+### Platform Setup
+
+<Callout>On iOS, there should be no additional setup steps.</Callout>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:341]:
```suggestion
On iOS, there are no additional setup steps.
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 2]
```diff
@@ -356,6 +420,242 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+</InlineFilter>
+
+<InlineFilter filters={['flutter']}>
+## How It Works
+
+Sign-in with web UI will display the sign-in UI inside a webview. After the sign-in process is complete, the sign-in UI will redirect back to your app.
+
+## Platform Setup
+
+<h3>Web</h3>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:432]:
```suggestion
### Web
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 3]
```diff
@@ -356,6 +420,242 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+To use Hosted UI in your Flutter web application locally, you must run the app with the `--web-port=3000` argument (with the value being whichever port you assigned to localhost host when configuring your redirect URIs).
+
+<h3>Android</h3>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:436]:
```suggestion
### Android
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 4]
```diff
@@ -356,6 +420,242 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+            <category android:name="android.intent.category.DEFAULT" />
+            <category android:name="android.intent.category.BROWSABLE" />
+            <data android:scheme="myapp" />
+        </intent-filter>
+  </activity>
+  ...
+</application>
+```
+
+<h3>macOS</h3>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:464]:
```suggestion
### macOS
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 5]
```diff
@@ -356,6 +420,242 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+Open XCode and enable the App Sandbox capability and then select "Incoming Connections (Server)" under "Network".
+
+![Incoming Connections setting selected in the App Sandbox section of the runner signing and capabilities tab.](/images/project-setup/flutter/mac/xcode-entitlements.png)
+
+<h3>iOS, Windows and Linux</h3>
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:470]:
```suggestion
### iOS, Windows and Linux
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 6]
```diff
@@ -356,6 +420,242 @@ When you import and use the `signInWithRedirect` function, it will add a listene
...
+
+No specific platform configuration is required.
+
+## Launch Social Web UI Sign In
+
+Sweet! You're now ready to launch sign in with your social provider's web UI.
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:476]:
```suggestion
You're now ready to launch sign in with your social provider's web UI.
```

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 7]
```diff
@@ -28,6 +28,26 @@ export function getStaticProps(context) {
   };
 }
 
+<InlineFilter filters={['flutter']}>
+## Prerequisites
+
+* An app setup according to the [getting started walkthrough](/[platform]/build-a-backend/auth/set-up-auth/)
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:34]: set up

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 8]
```diff
@@ -28,6 +28,26 @@ export function getStaticProps(context) {
...
+
+</InlineFilter>
+
+<InlineFilter filters={['swift']}>
+## Prerequisites
+
+* An app setup according to the [getting started walkthrough](/[platform]/build-a-backend/auth/set-up-auth/)
```
Comment (aspittel) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:48]: set up

**Review by aspittel — DISMISSED** [review]

**Review by renebrandel — APPROVED** [review]

File: src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx [hunk 9]
```diff
@@ -28,6 +28,26 @@ export function getStaticProps(context) {
...
+* An app set up according to the [getting started walkthrough](/[platform]/build-a-backend/auth/set-up-auth/)
+
+</InlineFilter>
+
+<InlineFilter filters={['swift']}>
+## Prerequisites
+
+* An app set up according to the [getting started walkthrough](/[platform]/build-a-backend/auth/set-up-auth/)
+
+</InlineFilter>
```
Comment (timngyn) [inline src/pages/[platform]/build-a-backend/auth/add-social-provider/index.mdx:50]: These look like the same links and text? It should be possible to just have one setup for all three of these filters like so if these links are supposed to be the same:

```suggestion
<InlineFilter filters={['flutter', 'android', 'swift']}>
## Prerequisites

* An app set up according to the [getting started walkthrough](/[platform]/build-a-backend/auth/set-up-auth/)

</InlineFilter>
```

**Review by josefaidt — APPROVED** [review]

**Pull request merged**